packages/neo-one-client-core/src/user/RemoteUserAccountProvider.ts

```diff
@@ -0,0 +1,273 @@
+    this.networksInternal$ = new BehaviorSubject([] as ReadonlyArray<NetworkType>);
+    this.messageEndpoint.addEventListener((event: Message) => {
+      if (event.type === 'OBSERVABLE') {
+        if (event.id === 'userAccounts$') {
+          this.accountsInternal$.next(event.value);
+        }
+        if (event.id === 'currentUserAccount$') {
+          this.currentAccountInternal$.next(event.value);
+        }
+        if (event.id === 'networks$') {
+          this.networksInternal$.next(event.value);
+        }
+      }
+    });
+    this.userAccounts$ = this.accountsInternal$;
+    this.currentUserAccount$ = this.currentAccountInternal$.pipe(distinctUntilChanged());
+    this.networks$ = this.networksInternal$;
+  }
+
+  public getCurrentUserAccount(): UserAccount | undefined {
+    return this.currentAccountInternal$.getValue();
+  }
+
+  public async selectUserAccount(userAccountID?: UserAccountID): Promise<void> {
+    return this.handleRelayMethod({ method: 'selectUserAccount', args: [userAccountID] });
+  }
+
+  public async deleteUserAccount(userAccountID?: UserAccountID): Promise<void> {
+    return this.handleRelayMethod({ method: 'deleteUserAccount', args: [userAccountID] });
+  }
+
+  public getUserAccounts(): ReadonlyArray<UserAccount> {
+    return this.accountsInternal$.getValue();
+  }
+
+  public getNetworks(): ReadonlyArray<NetworkType> {
+    return this.networksInternal$.getValue();
+  }
+
+  public async updateUserAccountName(options: UpdateAccountNameOptions): Promise<void> {
+    return this.handleRelayMethod({ method: 'updateUserAccountName', args: [options] });
+  }
+
+  public async getBlockCount(network: NetworkType): Promise<number> {
+    return this.handleRelayMethod({ method: 'getBlockCount', args: [network] });
+  }
+
+  public async getAccount(network: NetworkType, address: AddressString): Promise<Account> {
+    return this.handleRelayMethod({ method: 'getAccount', args: [network, address] });
+  }
+
+  public iterBlocks(network: NetworkType, options?: IterOptions): AsyncIterable<Block> {
+    return asyncGenerator(this.messageEndpoint, 'iterBlocks', network, options);
+  }
+
+  public iterActionsRaw(network: NetworkType, options?: IterOptions): AsyncIterable<RawAction> {
+    return asyncGenerator(this.messageEndpoint, 'iterActionsRaw', network, options);
+  }
+
+  public async transfer(
+    transfers: ReadonlyArray<Transfer>,
+    options?: TransactionOptions,
+  ): Promise<TransactionResult<TransactionReceipt, InvocationTransaction>> {
+    return this.handleRelayMethod({ method: 'transfer', args: [transfers, options] });
+  }
+
+  public async claim(options?: TransactionOptions): Promise<TransactionResult<TransactionReceipt, ClaimTransaction>> {
+    return this.handleRelayMethod({ method: 'claim', args: [options] });
+  }
+
+  public async invoke(
+    contract: AddressString,
+    method: string,
+    params: ReadonlyArray<ScriptBuilderParam | undefined>,
```

Comment:
Can you add a test (for this method and/or for other methods) that uses every possible `ScriptBuilderParam` and `Param`?